core/src/main/scala/com/microsoft/azure/synapse/ml/explainers/ICEFeature.scala

```diff
@@ -0,0 +1,168 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.azure.synapse.ml.explainers
+
+import spray.json._
+
+private[explainers] abstract class ICEFeature(val name: String, outputColName: Option[String] = None) {
+  def validate: Boolean
+  private val defaultOutputColName = name + "_dependence"
+  def getOutputColName: String = this.outputColName.getOrElse(defaultOutputColName)
+}
+
+/**
+  * Represents a single categorical feature to be explained by ICE explainer.
+  * @param name The name of the categorical feature.
+  * @param numTopValues The max number of top-occurring values to be included in the categorical feature.
+  *                     Default: 100.
+  * @param outputColName The name for output column with explanations for the feature.
+  *                      Default: input name of the feature + _dependence.
+  */
+case class ICECategoricalFeature(override val name: String, numTopValues: Option[Int] = None,
+                                 outputColName: Option[String] = None)
+ extends ICEFeature(name, outputColName) {
+  override def validate: Boolean = {
+    numTopValues.forall(_ > 0)
+  }
+
+  private val defaultNumTopValue = 100
+  def getNumTopValue: Int = {
+    this.numTopValues.getOrElse(defaultNumTopValue)
+  }
+}
+
+/**
+  * Companion object to provide JSON serializer and deserializer for ICECategoricalFeature.
+  */
+object ICECategoricalFeature {
+  implicit val JsonFormat: JsonFormat[ICECategoricalFeature] = new JsonFormat[ICECategoricalFeature] {
```

Comment:
Do the default case class serializers not work here?

Reply:
Oh I wasn't aware of that. Elena and I will look into that.